src/graph.jl

```diff
@@ -163,6 +163,61 @@ function all_node_properties(table::Shapefile.Table)::Array{Dict{String, Any}}
 end
 
 
+function get_attribute_by_key(node_attributes::Array,
+                              column_name::String,
+                              process_value::Function=identity)::Array
+    """ Returns an array whose values correspond to the value of an attribute
+        for each node.
+
+        Arguments:
+
+        node_attributes :   An array of Dict{String, Any}, where each dictionary
+                            represents a mapping from attribute names to values
+                            for a particular node.
+        column_name     :   The name of the attribute (i.e., the key of the
+                            attribute in the dictionaries)
+        process_value   :   An optional argument that processes the raw value
+    """
+    return [process_value(n[column_name]) for n in node_attributes]
+end
+
+
+function population_to_int(raw_value::Number)::Int
+    """ Tiny helper function to coerce population counts (whether they are
+        ints or floats) to int.
+    """
+    return raw_value isa Int ? raw_value : convert(Int, round(raw_value))
+end
+
+
+function get_assignments(node_attributes::Array,
+                         assignment_col::AbstractString)::Array{Int, 1}
+    """ Extracts the assignments for each node in a graph. First attempts
+        to parse the values of the assignments column as an Int; if it fails,
+        it assumes every unique string assignment corresponds to to a unique
+        district.
+    """
+    assignment_to_num = Dict{String, Int}() # map unique strings to integers
+    raw_assignments = get_attribute_by_key(node_attributes, assignment_col)
+    processed_assignments = zeros(length(raw_assignments))
+    for (i, raw_value) in enumerate(raw_assignments)
+        if raw_value isa Int
+            processed_assignments[i] = raw_value
+        elseif raw_value isa String
+            try
+                processed_assignments[i] = parse(Int, raw_assignment)
+            catch exception # if the String could not be read as an int
```

Comment:
Too generic?

Reply:
i don't think so! i didn't like how we had arbitrarily grouped together `get_populations_and_assignments` before. disaggregating them and using this more general method made more sense to me, especially because they involve slightly different ways of processing the data.